src/gc.cr

```diff
@@ -1,13 +1,76 @@
+# :nodoc:
+fun __crystal_malloc(size : UInt32) : Void*
+  GC.malloc(LibC::SizeT.new(size))
+end
+
+# :nodoc:
+fun __crystal_malloc_atomic(size : UInt32) : Void*
+  GC.malloc(LibC::SizeT.new(size))
+end
+
+# :nodoc:
+fun __crystal_realloc(pointer : Void*, size : UInt32) : Void*
+  GC.realloc(pointer, LibC::SizeT.new(size))
+end
+
+# :nodoc:
+fun __crystal_malloc64(size : UInt64) : Void*
+  {% if flag?(:bits32) %}
+    if size > UInt32::MAX
+      raise ArgumentError.new("Given size is bigger than UInt32::MAX")
+    end
+  {% end %}
+
+  GC.malloc(LibC::SizeT.new(size))
+end
+
+# :nodoc:
+fun __crystal_malloc_atomic64(size : UInt64) : Void*
+  {% if flag?(:bits32) %}
+    if size > UInt32::MAX
+      raise ArgumentError.new("Given size is bigger than UInt32::MAX")
+    end
+  {% end %}
+
+  GC.malloc(LibC::SizeT.new(size))
+end
+
+# :nodoc:
+fun __crystal_realloc64(ptr : Void*, size : UInt64) : Void*
+  {% if flag?(:bits32) %}
+    if size > UInt32::MAX
+      raise ArgumentError.new("Given size is bigger than UInt32::MAX")
+    end
+  {% end %}
+
+  GC.realloc(ptr, LibC::SizeT.new(size))
+end
+
 module GC
+  record Stats,
+    # collections : LibC::ULong,
+    # bytes_found : LibC::Long,
+    heap_size : LibC::ULong,
+    free_bytes : LibC::ULong,
+    unmapped_bytes : LibC::ULong,
+    bytes_since_gc : LibC::ULong,
+    total_bytes : LibC::ULong
+
   def self.malloc(size : Int)
-    __crystal_malloc(size.to_u32)
+    malloc(LiC::SizeT.new(size))
   end
 
   def self.malloc_atomic(size : Int)
-    __crystal_malloc_atomic(size.to_u32)
+    malloc_atomic(LibC::SizeT.new(size))
   end
 
   def self.realloc(pointer : Void*, size : Int)
-    __crystal_realloc(pointer, size.to_u32)
+    realloc(pointer, LibC::SizeT.new(size))
   end
 end
+
+{% if flag?(:GC_NULL)%}
```

Comment:
Typically flags so far have been lowercase so far (`without_openssl`, etc), why not `null_gc`?

Reply:
Or maybe `-Dno_gc`?

Reply:
If that's the case, maybe we should rename `gc/null` to `gc/none`?

Reply:
I wouldn’t use “no gc”. Seems more like a toggle to switch to a c++ like memory management to me.

Reply:
Habit? Too much C reading? Constant-like looking? Anyway, can change to whatever.